src/query/query_opfunc.c

```diff
@@ -12839,13 +12840,30 @@ qdata_load_agg_hvalue_in_agg_list (AGGREGATE_HASH_VALUE * value, AGGREGATE_TYPE
 	      /* set tuple count */
 	      agg_list->accumulator.curr_cnt = value->accumulators[i].curr_cnt;
 
-	      /* shallow copy dbval */
+	      /*
+	       * shallow, fast copy dbval. This may be unsafe. Internally, value->accumulators[i].value and
+	       * agg_list->accumulator.value values keeps the same pointer to a buffer. If a value is cleared, the other
+	       * value refer a invalid memory. Probably a safety way would be to use clone.
+	       */
 	      *(agg_list->accumulator.value) = *(value->accumulators[i].value);
 	      *(agg_list->accumulator.value2) = *(value->accumulators[i].value2);
 
-	      /* mark as container */
+	      /* reset accumulator values. */
 	      value->accumulators[i].value->need_clear = false;
+	      db_type = DB_VALUE_DOMAIN_TYPE (value->accumulators[i].value);
+	      if (db_type == DB_TYPE_VARNCHAR || db_type == DB_TYPE_VARCHAR
+		  || db_type == DB_TYPE_CHAR || db_type == DB_TYPE_NCHAR)
```

Comment:
I'd like to propose a change:
```c
if (db_type == DB_TYPE_VARCHAR || db_type == DB_TYPE_VARNCHAR)
```

since fixed chars are irrelevant to string compression and VARCHAR is dominant, which is better to precede VARNCHAR.